src/SFML/Graphics/Sprite.cpp

```diff
@@ -36,19 +36,15 @@
 namespace sf
 {
 ////////////////////////////////////////////////////////////
-Sprite::Sprite(const Texture& texture)
+Sprite::Sprite(const Texture& texture) : Sprite(texture, IntRect({0, 0}, Vector2i(texture.getSize())))
 {
-    setTexture(texture, true);
 }
 
 
 ////////////////////////////////////////////////////////////
-Sprite::Sprite(const Texture& texture, const IntRect& rectangle)
+Sprite::Sprite(const Texture& texture, const IntRect& rectangle) : m_texture(&texture), m_textureRect(rectangle)
 {
-    // Compute the texture area
-    setTextureRect(rectangle);
-    // Assign texture
-    setTexture(texture, false);
+    updateVertices();
```

Comment:
Before this PR `updateVertices()` was not called when using the `const Texture& texture` overload. It was also not called when the 2-parameter was called with a zero-size rectangle. Now `updateVertices` is called unconditionally. This is not incorrect but it's a bit unnecessary. Is it worth adding an `if (m_textureRect != IntRect())` conditional here to skip calling `updateVertices` when it is not needed?

Reply:
You raise a good point, however... when would you ever create a zero-sized sprite?

Now that `Texture` is guaranteed to be non-empty due to #3027, I would expect `texture.getSize()` on L39 to never be `{0, 0}`, right?

Furthermore, what would be the point of creating a sprite with a texture and a zero-sized rectangle?

It seems to me that we were "optimizing" for a very niche scenario, and that such "optimization" might not even be one as it's an additional branch.

I agree with your reasoning in this comment https://github.com/SFML/SFML/pull/3026#issuecomment-2123624265 -- the complexity added by an extra branch like the one you're describing would need to be motivated by a compelling benchmark, and I don't think that benchmarking the creation of zero-sized sprites makes much sense, unless I am missing something...

Reply:
@ChrisThrasher: are you OK to resolve this comment? 

I also think that the whole "store `m_textureRect` and check if we need to recompute the vertices" is very unnecessary, it's a micro-optimization that might even be dentrimental to performance due to branching, and we're talking about `sf::Sprite` here which is already one of the slowest ways you could draw a quad to the screen.